zmon-controller-ui/grafana/public/plugins/grafana-kairosdb-datasource/datasource.js

```diff
@@ -125,16 +136,23 @@ function (angular, _, sdk, dateMath, kbn) {
     });
   };
 
-  KairosDBDatasource.prototype._performMetricKeyValueLookup = function(metric, key) {
+  KairosDBDatasource.prototype._performMetricKeyValueLookup = function(metric, key, value) {
     if(!metric || !key) {
       return this.q.when([]);
     }
 
+    var _metric = { name: metric };
+
+    if (key && value) {
+        _metric.tags = {};
+        _metric.tags[key] = [ value ];
```

Comment:
it doesn't take into account the previous tags, just the metric and the currenty key, since its not a query but just an autocomplete on tags.